PythonVisualizations/SimpleQueue.py

```diff
@@ -230,133 +264,127 @@ def display(self):
+        self.window.update()
 
     #disable insert if queue if full, disable delete if empty
     #enable everything else without overriding queue/deque functionality
     def onOffButtons(self):
         #if it's a deque
-        if buttons[5]['relief'] == 'sunken':
-            enableButtons()
+        if self.buttons[5]['relief'] == 'sunken':
+            pass
+          #  enableButtons()
             
         #if it's a queue
         else:
-            buttons[0].config(state = NORMAL)
-            buttons[3].config(state = NORMAL)
+            self.buttons[0].config(state = NORMAL)
+            self.buttons[3].config(state = NORMAL)
         
         #in all cases: disable buttons as necessary
         if self.nItems == self.size:
-            buttons[0].config(state = DISABLED)
-            buttons[1].config(state = DISABLED)
+            self.buttons[0].config(state = DISABLED)
+            self.buttons[1].config(state = DISABLED)
             
         if self.nItems == 0:
-            buttons[2].config(state = DISABLED)
-            buttons[3].config(state = DISABLED)
+            self.buttons[2].config(state = DISABLED)
+            self.buttons[3].config(state = DISABLED)
 
-def pause(pauseButton):
-    global waitVar
-    waitVar.set(True)
 
-    pauseButton['text'] = "Play"
-    pauseButton['command'] = lambda: onClick(play, pauseButton)
 
-    canvas.wait_variable(waitVar)
-
-def play(pauseButton):
-    global waitVar
-    waitVar.set(False)
-
-    pauseButton['text'] = 'Pause'
-    pauseButton['command'] = lambda: onClick(pause, pauseButton)
-
-def onClick(command, parameter = None):
-    cleanUp()
-    #disableButtons()
-    if parameter:
-        command(parameter)
-    else:
-        command()
-    #if command not in [pause, play]:
-     #   enableButtons()
-
-def cleanUp():
-    global cleanup
-    if len(cleanup) > 0:
-        for o in cleanup:
-            canvas.delete(o)
-    outputText.set('')
-    window.update()
+    def onClick(self,command, parameter = None):
+        self.cleanUp()
+        #disableButtons()
+        if parameter:
+            command(parameter)
+        else:
+            command()
+        #if command not in [pause, play]:
+         #   enableButtons()
 
 # Button functions
 
-def clickInsertRear():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            queue.insertRear(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-        textBox.delete(0, END )
+    def clickInsertRear(self):
+        entered_text = self.textEntries[0].get()
+        if entered_text:
+            val = int(entered_text)
+            if val < 100:
+                self.insertRear(int(entered_text))
+            else:
+                self.outputText.set("Input value must be an integer from 0 to 99.")
+            self.textEntries[0].delete(0, END )
```

Comment:
Please use the methods of `VisualizationApp` designed for managing operational arguments.  Specifically, use `self.getArgument()` to get the text entered by the user and `self.clearArgument()` to clear the entry.  Doing so enables the class to manage the state of buttons dependent on those arguments.